Add anchor underline to high contrast

dancormier opened the pull request:
Upstream changes to support https://github.com/StackEng/StackOverflow/pull/9535

Only merge once https://github.com/StackEng/StackOverflow/pull/9535 and https://github.com/StackEng/StackOverflow/pull/9526 are merged.

dancormier committed:
Add anchor underline in high contrast

See also:
- https://github.com/StackEng/StackOverflow/pull/9535
- https://github.com/StackEng/StackOverflow/pull/9526

Addresses: https://meta.stackoverflow.com/a/411839/652353

aaronshekey commented on lib/css/base/_stacks-reset-normalize.less:
```diff
@@ -99,11 +99,14 @@ pre {
   background-color: transparent; /* 1 */
   -webkit-text-decoration-skip: objects; /* 2 */
+
+  .highcontrast-mode({ text-decoration: underline; });
```
Let's do this in `stacks-links.less` instead: https://github.com/StackExchange/Stacks/blob/develop/lib/css/components/_stacks-links.less#L17

There's a future where we kill these resets entirely #609

dancormier replied:
Good call. I knew it was wrong when I was doing it and I did it anyway. 🤷‍♂️

dancormier committed:
Move anchor underline to _stacks-links.less

Addresses: https://github.com/StackExchange/Stacks/pull/769#discussion_r723684016

aaronshekey commented on lib/css/base/_stacks-reset-normalize.less:
```diff
@@ -99,6 +99,7 @@ pre {
 /**
  * 1. Remove the gray background on active links in IE 10.
  * 2. Remove gaps in links underline in iOS 8+ and Safari 8+.
+ * 3. Adds underline to links in high contrast
```
```suggestion
```

aaronshekey reviewed (changes requested):
Just 2 teeny tiny bits. Merge whenever you'd like into production SO. That work can definitely ship before the next release of Stacks.

We've got a blue that's a little leaky in the case of `s-card`. 
<img width="320" alt="image" src="https://user-images.githubusercontent.com/1369864/137968582-ad0f08bd-4a53-40a7-9078-f85b72b42a96.png">

dancormier committed:
Add code comment mentioning hc underline

Co-authored-by: Aaron Shekey <aaron@aaronshekey.com>

aaronshekey committed:
Try simplifying the link definition

aaronshekey committed:
Address card links
